### ISSUE TITLE
Top Skill Extraction Issues

### ISSUE THREAD

**Issue by alexhendrik** [workforce-data-initiative/skills-ml#280]
I apologize in advance as this is probably not the right place for a question like this, but I am getting somewhat desperate and cannot find help on any separate resources.  

I have been trying to use skills-ml to parse a large dataset of job postings that I store in JSON, and I managed to make it work except for one part: I can't get it to show me the most common skills that were mentioned within the dataset.  I have even tried modifying the existing functions from skill_extractors, but all I can get is the number of unique skills found by each skill extractor.  

Could you please point me in the right direction?  I could not find an answer to this in the documentation either. Thank you in advance!

**Comment by thcrock** [comment]
What code are you using? For what you're describing, it sounds like you may just be able to use `document_skill_counts`. I'd suggest something similar to this example:

https://github.com/workforce-data-initiative/skills-ml/blob/master/examples/NounPhraseSkillExtraction.py

In short, the skill extractor instance should have a 'document_skill_counts' method that returns a collections.Counter of skill->count that you can add together for the whole dataset.

**Comment by alexhendrik** [comment]
Thank you so much! I was looking in the wrong direction before and had assumed that document_skill_counts simply referred to the number of skills detected without specifying the skill occurrence itself.  

However, I have hit the following issue when implementing it very closely based on the example:

> `Traceback (most recent call last):
  File "c:\Users\Alexh\Desktop\VSCode Workspace\JobParser\SkillExtractor.py", line 151, in <module>
    skill_counts += skill_extractor.document_skill_counts(job_posting)
  File "C:\Program Files\Python36\lib\site-packages\skills_ml-2.1.0-py3.6.egg\skills_ml\algorithms\skill_extractors\base.py", line 153, in document_skill_counts
    for candidate_skill in self.candidate_skills(source_object):
  File "C:\Program Files\Python36\lib\site-packages\skills_ml-2.1.0-py3.6.egg\skills_ml\algorithms\skill_extractors\section_extract.py", line 31, in candidate_skills
    lines_in_section = section_extract(self.section_regex, source_object['description'])
TypeError: byte indices must be integers or slices, not str`

I have seen similar byte-related issues when using your existing functions, so I'll try to figure it out and update you on any findings.

Thank you so much for the pointer again, it is rare for library developers to be willing to help like this!

**Comment by alexhendrik** [comment]
After some more testing, I have found out that the problem lies within my own dataset, since it seems to work fine with the JobCollectionSample().  Thank you again for your help!

**Comment by thcrock** [comment]
No problem! At first glance, it looks like you may just have to convert your data into a dictionary with json.loads(data.decode('utf-8')) before passing it to the skill extractor. If I recall correctly, the old skill extractor interface took JSON strings, but we at some point switched to the pure object as a dictionary and I'm not sure we updated all the docs so you may have been following a doc we haven't updated.

**Comment by alexhendrik** [comment]
I managed to get it working last night, my (very poor) solution was to mimic the Virginia format with my data and the error was caused by my code running the collection of jobs through a sample factory for seemingly no reason and passing that into the counter loop, I guess it was just legacy code that I forgot to take out.  Without the sample factory it worked flawlessly!  I really appreciate the help!
